sdk/src/main/java/software/amazon/awssdk/iot/discovery/DiscoveryClient.java

```diff
@@ -120,5 +137,9 @@ public void close() {
         if(httpClientConnectionManager != null) {
             httpClientConnectionManager.close();
         }
+        if (cleanExecutor == true && executorService != null) {
```

Comment:
I don't think we need the executorService != null check unless there's some way that the executorService can be shutdown and made null outside of this close() after having been created specifically for this client.